cli/src/commands/mod.rs

```diff
@@ -1061,13 +1061,8 @@ fn cmd_restore(
         from_tree.id().clone()
     } else {
         let matcher = workspace_command.matcher_from_values(&args.paths)?;
-        let mut tree_builder = MergedTreeBuilder::new(to_commit.tree_id().clone());
         let to_tree = to_commit.tree()?;
-        for (repo_path, diff) in from_tree.diff(&to_tree, matcher.as_ref()) {
-            let (before, _after) = diff?;
-            tree_builder.set_or_remove(repo_path, before);
-        }
-        tree_builder.write_tree(workspace_command.repo().store())?
+        restore_tree(&from_tree, &to_tree, matcher.as_ref())?
```

Comment:
nit: `if args.paths.is_empty()` can be removed since `matcher_from_values()` will return `EverythingMatcher` in that case.

Reply:
Done.